### PR TITLE
Add unit tests for functions in cni/ovs.go

### PR DESCRIPTION
**Pull request by sheetalsingala**
Signed-off-by: sheetalsingala <ssingala@redhat.com>

**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: go-controller/pkg/cni/ovs_unit_test.go [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+	"fmt"
+	"testing"
+
+	mock_k8s_io_utils_exec "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing/mocks/k8s.io/utils/exec"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	kexec "k8s.io/utils/exec"
+)
+
+type onCallReturnArgs struct {
```
Comment (vjayaramrh) [inline go-controller/pkg/cni/ovs_unit_test.go:13]: you may want to use the struct definition at https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/testing/testifymockhelper.go#L3

File: go-controller/pkg/cni/ovs_unit_test.go [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+				mockCall.Once()
+			}
+
+			runner = tc.runnerInstance
+
+			_, e := ovsExec()
+
+			if tc.expectedErr != nil {
+				assert.Error(t, e)
+			}
```
Comment (vjayaramrh) [inline go-controller/pkg/cni/ovs_unit_test.go:106]: maybe better to add else statement as well to ensure that the error returned would be `nil`
`assert.Nil(t, e)`

File: go-controller/pkg/cni/ovs_unit_test.go [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+				mockCall.Once()
+			}
+
+			runner = tc.runnerInstance
+
+			_, e := ovsCreate("blah")
+
+			if tc.expectedErr != nil {
+				assert.Error(t, e)
+			}
```
Comment (vjayaramrh) [inline go-controller/pkg/cni/ovs_unit_test.go:162]: check for `assert.Nil(t, e)` in the else statement

File: go-controller/pkg/cni/ovs_unit_test.go [hunk 4]
```diff
@@ -0,0 +1,406 @@
...
+				mockCall.Once()
+			}
+
+			runner = tc.runnerInstance
+
+			e := ovsDestroy("table", "record")
+
+			if tc.expectedErr != nil {
+				assert.Error(t, e)
+			}
```
Comment (vjayaramrh) [inline go-controller/pkg/cni/ovs_unit_test.go:218]: check for assert.Nil(t, e) in the else statement

File: go-controller/pkg/cni/ovs_unit_test.go [hunk 5]
```diff
@@ -0,0 +1,406 @@
...
+				mockCall.Once()
+			}
+
+			runner = tc.runnerInstance
+
+			e := ovsSet("table", "record")
+
+			if tc.expectedErr != nil {
+				assert.Error(t, e)
+			}
```
Comment (vjayaramrh) [inline go-controller/pkg/cni/ovs_unit_test.go:274]: check for assert.Nil(t, e) in the else statement

File: go-controller/pkg/cni/ovs_unit_test.go [hunk 6]
```diff
@@ -0,0 +1,406 @@
...
+				mockCall.Once()
+			}
+
+			runner = tc.runnerInstance
+
+			_, e := ovsFind("table", "record", "condition")
+
+			if tc.expectedErr != nil {
+				assert.Error(t, e)
+			}
```
Comment (vjayaramrh) [inline go-controller/pkg/cni/ovs_unit_test.go:344]: check for assert.Nil(t, e) in the else statement

File: go-controller/pkg/cni/ovs_unit_test.go [hunk 7]
```diff
@@ -0,0 +1,406 @@
...
+	tests := []struct {
+		desc                string
+		expectedErr         error
+		onRetArgsKexecIface *ovntest.TestifyMockHelper
+		onRetArgsCmdList    *ovntest.TestifyMockHelper
+		runnerInstance      kexec.Interface
+	}{
+		{
+			desc:                "Test codepath when ovsExec returns an error",
+			expectedErr:         nil,
```
Comment (vjayaramrh) [inline go-controller/pkg/cni/ovs_unit_test.go:295]: per the test description, there is an indication that error is returned, but on L295 the expectedErr is set to nil, do you mind clariflying? Thanks

File: go-controller/pkg/cni/ovs_unit_test.go [hunk 8]
```diff
@@ -0,0 +1,406 @@
...
+				mockCall.Once()
+			}
+
+			runner = tc.runnerInstance
+
+			e := ovsClear("table", "record", "columns")
+
+			if tc.expectedErr != nil {
+				assert.Error(t, e)
+			}
```
Comment (vjayaramrh) [inline go-controller/pkg/cni/ovs_unit_test.go:400]: check for assert.Nil(t, e) in the else statement

File: go-controller/pkg/cni/ovs_unit_test.go [hunk 9]
```diff
@@ -0,0 +1,406 @@
...
+
+	tests := []struct {
+		desc                string
+		expectedErr         error
+		onRetArgsKexecIface *ovntest.TestifyMockHelper
+		onRetArgsCmdList    *ovntest.TestifyMockHelper
+		runnerInstance      kexec.Interface
+	}{
+		{
+			desc:           "Test codepath when runner is nil and returns error",
```
[the remark this answers is not in the document]
Comment (sheetalsingala) [reply go-controller/pkg/cni/ovs_unit_test.go:58]: I missed that. Thanks

**Comment by vjayaramrh** [comment]
@girishmg @trozet 
/lgtm

**Comment by vjayaramrh** [comment]
@sheetalsingala thanks for addressing the review comments, looks like 2 new functions have been added in the last 2 days that have been merged into the master.
Refer https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/cni/ovs.go#L74 and https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/cni/ovs.go#L101 for which tests may need to be added

**Comment by vjayaramrh** [comment]
@trozet @girishmg 
/lgtm

**Pull request merged**